File: 693. Binary Number with Alternating Bits.cpp
class Solution {
public:
    bool hasAlternatingBits(int n) {
        int preval = 2;
        while(n!=0){
            int val = n & 1;
            if(val==preval) return false;
            preval = val;
            n>>=1;
        }
        return true;
    }
};

File: 693. Binary Number with Alternating Bits.js
/**
 * @param {number} n
 * @return {boolean}
 */
var hasAlternatingBits = function(n) {
    let preval = 2;
    while(n!==0){
        let val = n &1;
        if(preval==val) return false;
        preval = val;
        n >>= 1;
    }
    return true;
};